addOns/pscanrulesBeta/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/InformationDisclosureSuspiciousComments.java

```diff
@@ -45,25 +45,67 @@
     private static final int PLUGIN_ID = 10027;
 
     private PassiveScanThread parent = null;
-    private static final String databaseErrorFile = "xml/suspicious-comments.txt";
+    public static final String suspiciousCommentsListDir = "xml";
+    public static final String suspiciousCommentsListFile = "suspicious-comments.txt";
     private static final Logger logger =
             Logger.getLogger(InformationDisclosureSuspiciousComments.class);
 
-    private List<Pattern> patterns = null;
+    private static final List<Pattern> PATTERNS;
+
+    /*
+     * Static initializer to load the suspicious comments list from the file system
+     * and put the results into the PATTERNS List.
+     */
+    static {
```

Comment:
Why this change? This means the file will be loaded/read even if the scanner is disabled/unused.

Reply:
Only if the class is loaded, and who cares? The way it was written before it was a non-static method/variable so if you created multiple scanner instances, this got loaded for each instance, when they would all be the same, so in that scenario there would be less loads done. And this implementation is far simpler than the previous setup anyway. This seems like a pretty picky comment to me.

Reply:
> Only if the class is loaded

The class will be loaded always, passive scanners are loaded during start up.

> The way it was written before it was a non-static method/variable so if you created multiple scanner instances, this got loaded for each instance, when they would all be the same, so in that scenario there would be less loads done.

That's unrelated to my question, it was added to the static block not on the variable. (fyi, passive scan just uses one instance so that case would not happen anyway.)

> And this implementation is far simpler than the previous setup anyway.

The implementation should be the same but done lazily as before (keeping it in a method does not make it complex).

> This seems like a pretty picky comment to me.

Is it picky wanting to know why the existing behaviour was changed? I don't read minds (yet?) so I can only ask (sometimes it's clear and obvious why a change was made for the better others aren't).

Reply:
We have no idea when/how these scanners are loaded nor how many times they are instantiated. You can't tell from the code. So with a static initializer you know it gets initialized exactly once. As such, I recommend leaving it as is. We won't change any of the other scanners now that we know how things work. That OK?

Reply:
Changes should only be made for a good reason, not just because you dont know how something works :)
In this case @thc202 has given a good reason as to why it should not be static, eg if the rule is disabled. I think thats a very good reason to leave it as is.
@thc202's reviews might seem to be 'picky' but tbh I rely on them being so thorough.
He finds 'fault' with my code all of the time, and I always use that as an opportunity to learn :)

Reply:
It would be better to keep the old behaviour for the reasons stated earlier. To be clear I'm not (and wasn't) asking to change this to an instance variable, it makes sense and it's more clear to keep it as a class variable, just load the file lazily as before (it's just "cut/paste" to a class method).